### PR TITLE
ODBC driver: alter-test

### PR DESCRIPTION
**Pull request by tiagokepe**
This PR provides the support for the alter-test and also enables the clang-tidy on the ODBC code.

### CODE REVIEW

File: tools/odbc/include/api_info.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	static const std::vector<SQLSMALLINT> ODBC_SUPPORTED_SQL_TYPES;
+
+	struct LogicalTypeCmp {
+		bool operator()(const LogicalTypeId a, const LogicalTypeId b) const {
+			return a < b;
+		}
+	};
+	// Map DuckDB LogicalTypes to ODBC SQL Types
+	static const std::map<duckdb::LogicalTypeId, SQLSMALLINT, LogicalTypeCmp> MAP_SQL_TYPES;
```
Comment (Mytherin) [inline tools/odbc/include/api_info.hpp:33]: Could this map be moved into the `FindRelatedSQLType` function and turned into a `switch` statement?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. One minor comment, but otherwise I think it's ready to merge.

File: tools/odbc/include/api_info.hpp (hunk 1 above)
Comment (tiagokepe) [reply tools/odbc/include/api_info.hpp]: Done. CI is in progress now. ;)

**Comment by Mytherin** [comment]
Thanks, looks good!

**Pull request merged**